kubeadm/v1.15.0/KubeCluster.ps1

```diff
@@ -0,0 +1,288 @@
+    wget $Path -o $tmpPath
+    ipmo $tmpPath  -DisableNameChecking
+    Remove-Item $tmpPath
+}
+###############################################################################################
+# Download pre-req scripts
+
+$helperPath = "https://raw.githubusercontent.com/kubernetes-sigs/sig-windows-tools/kubeadm/v1.15.0/KubeClusterHelper.psm1"
+$helperDestination = "$PSScriptRoot\KubeClusterHelper.psm1" 
+wget $helperPath -o $helperDestination
+ipmo $helperDestination
+
+$hnsPath = "https://raw.githubusercontent.com/Microsoft/SDN/master/Kubernetes/windows/hns.psm1"
+$hnsDestination = "$PSScriptRoot\hns.psm1" 
+wget $hnsPath -o $hnsDestination
+ipmo $hnsDestination
+
+
+ReadKubeclusterConfig -ConfigFile $ConfigFile
+InitHelper
+PrintConfig
+WriteKubeClusterConfig
+
+
+# Initialize internal network modes of windows corresponding to 
+# the plugin used in the cluster
+$Global:NetworkName = "cbr0"
+$Global:NetworkMode = "l2bridge"
+if ($Global:NetworkPlugin -eq "vxlan")
+{
+    $Global:NetworkMode = "overlay"
+    $Global:NetworkName = "vxlan0"
+}
+######################################################################################################################
+
+# Handle --help
+if ($help.IsPresent)
+{
+    Usage
+    exit
+}
+
+# Handle --init
+if ($init.IsPresent)
+{
+    Write-Host "Initalizing a Windows Control Plane node is unsupported"
+    exit
+}
+
+# Handle --InstallPrerequisites
+if ($InstallPrerequisites.IsPresent)
+{
+    InstallContainersRole
+    $defaultValue = "Y"
+    if (!(Test-Path $env:HOMEDRIVE/$env:HOMEPATH/.ssh/id_rsa.pub))
+    {
+        $res = Read-Host "Do you wish to generate a SSH Key & Add it to the Linux Master [Y/n] - Default [Y] : "
+        if ($res -eq '' -or $res -eq 'Y'  -or $res -eq 'y')
+        {
+            ssh-keygen.exe
+        }
+    }
+
+    $pubKey = Get-Content $env:HOMEDRIVE/$env:HOMEPATH/.ssh/id_rsa.pub
+    Write-Host "Execute the below commands on the Linux Master($Global:MasterIp) to add this Windows Node's public key to its authorized keys"
+    
+    Write-Host "touch ~/.ssh/authorized_keys"
+    Write-Host "echo $pubKey >> ~/.ssh/authorized_keys"
+
+    $res = Read-Host "Continue to Reboot the host [Y/n] - Default [Y] : "
+    if ($res -eq '' -or $res -eq 'Y'  -or $res -eq 'y')
+    {
+        Restart-Computer -Force
+    }
+
+    InstallCRI $Global:Cri
+    InstallKubernetesBinaries -Destination  $Global:BaseDir -Source $Global:ClusterConfiguration.Kubernetes.Source
+
+    exit
+}
+
+# Handle -Join
+if ($Join.IsPresent)
+{
+    $kubeConfig = GetKubeConfig
+    if (!(KubeConfigExists))
+    {
+        # Fetch KubeConfig from the master
+        DownloadKubeConfig -Master $Global:MasterIp -User $Global:MasterUsername
+        if (!(KubeConfigExists))
+        {
+            throw $kubeConfig + " does not exist. Cannot connect to the master cluster"
+        }
+    }
+
+    # Validate connectivity with Master API Server
+
+    Write-Host "Trying to connect to the Kubernetes master"
```

Comment:
just a FYI, usage of "control-plane node" is preferred over "master" in user facing text.
there were debates about this.

seems to apply to multiple locations in the scripts.